cartography/intel/aws/iam.py

```diff
@@ -208,10 +208,9 @@ def load_roles(session, roles, current_aws_account_id, aws_update_tag):
         for statement in role["AssumeRolePolicyDocument"]["Statement"]:
             principal = statement["Principal"]
             principal_values = []
-            if 'AWS' in principal:
-                principal_type, principal_values = 'AWS', principal['AWS']
-            elif 'Service' in principal:
-                principal_type, principal_values = 'Service', principal['Service']
+            for t in ['AWS', 'Service', 'Federated']:
+                if t in principal:
```

Comment:
I understand what this code does but I feel like there's a neater way to do this without the `if` clause.  Are you able to revise?

Reply:
Sure! I did it!